path/spanning_tree.go

```diff
@@ -5,101 +5,166 @@
 package path
 
 import (
+	"container/heap"
+	"math"
 	"sort"
 
 	"github.com/gonum/graph"
 	"github.com/gonum/graph/concrete"
-	"github.com/gonum/graph/internal"
 )
 
-// EdgeListerGraph is an undirected graph than returns its complete set of edges.
-type EdgeListerGraph interface {
+// UndirectedWeighter is an undirected graph that returns distinct edge weights.
+type UndirectedWeighter interface {
 	graph.Undirected
-	Edges() []graph.Edge
+	graph.Weighter
 }
 
 // Prim generates a minimum spanning tree of g by greedy tree extension, placing
-// the result in the destination. The destination is not cleared first.
-func Prim(dst graph.MutableUndirected, g EdgeListerGraph) {
-	var weight Weighting
-	if wg, ok := g.(graph.Weighter); ok {
-		weight = wg.Weight
-	} else {
-		weight = UniformCost(g)
+// the result in the destination. The destination is not cleared first. The weight
```

Comment:
I'll meet you half way: "... in the destination, dst. The destination is not cleared first." I don't like "dst. dst".